src/test/java/org/springframework/integration/kafka/config/xml/KafkaMessageDrivenChannelAdapterParserTests.java

```diff
@@ -60,4 +70,57 @@ public void testKafkaMessageDrivenChannelAdapterParser() throws Exception {
 		assertThat(container).isNotNull();
 	}
 
+	@Test
+	@SuppressWarnings("unchecked")
+	public void testKafkaMessageDrivenChannelAdapterOptions() {
+		DefaultKafkaConsumerFactory<Integer, String> cf =
+				new DefaultKafkaConsumerFactory<>(Collections.<String, Object>emptyMap());
+		ContainerProperties containerProps = new ContainerProperties("foo");
+		KafkaMessageListenerContainer<Integer, String> container =
+				new KafkaMessageListenerContainer<>(cf, containerProps);
+		KafkaMessageDrivenChannelAdapter<Integer, String> adapter = new KafkaMessageDrivenChannelAdapter<>(container);
+		adapter.setOutputChannel(new QueueChannel());
+
+		adapter.setRecordFilterStrategy(mock(RecordFilterStrategy.class));
+		adapter.afterPropertiesSet();
+
+		Object messageListener = containerProps.getMessageListener();
```

Comment:
Yikes - this is a bug in s-k - we should copy the container properties in the ctor - we should not be able to modify the original props passed in by the user.